lib/cloud/test/redshift_serverless.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/redshiftserverless"
+	"github.com/aws/aws-sdk-go/service/redshiftserverless/redshiftserverlessiface"
+	"github.com/gravitational/trace"
+)
+
+// RedshiftServerlessMock mocks RedshiftServerless API.
+type RedshiftServerlessMock struct {
+	redshiftserverlessiface.RedshiftServerlessAPI
+
+	Workgroups           []*redshiftserverless.Workgroup
+	Endpoints            []*redshiftserverless.EndpointAccess
+	TagsByARN            map[string][]*redshiftserverless.Tag
+	GetCredentialsOutput *redshiftserverless.GetCredentialsOutput
+}
+
+func (m RedshiftServerlessMock) GetWorkgroupWithContext(_ aws.Context, input *redshiftserverless.GetWorkgroupInput, _ ...request.Option) (*redshiftserverless.GetWorkgroupOutput, error) {
+	for _, workgroup := range m.Workgroups {
+		if aws.StringValue(workgroup.WorkgroupName) == aws.StringValue(input.WorkgroupName) {
+			return new(redshiftserverless.GetWorkgroupOutput).SetWorkgroup(workgroup), nil
+		}
+	}
+	return nil, trace.NotFound("not found")
+}
+func (m *RedshiftServerlessMock) GetEndpointAccessWithContext(_ aws.Context, input *redshiftserverless.GetEndpointAccessInput, _ ...request.Option) (*redshiftserverless.GetEndpointAccessOutput, error) {
```

Comment:
nit: 
Inconsistency in mixed value/pointer `RedshiftServerlessMock` receiver. 
This is only method that uses `m *RedshiftServerlessMock` receivers but the logic doesn't differ from other Get/List functions.